exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/UnsupportedOperatorsVisitor.java

```diff
@@ -256,14 +255,6 @@ public SqlNode visit(SqlCall sqlCall) {
             "See Apache Drill JIRA: DRILL-1986");
         throw new UnsupportedOperationException();
       }
-
-      // Block Cross Join
-      if(join.getJoinType() == JoinType.CROSS) {
```

Comment:
Just to for the clarification is that for implicit cross joins and explicit cross joins do we have joinType set to CROSS by the calcite. If not then can we not distinguish them and only enable the explicit CROSS joins all the time??

Reply:
Yes, at this moment we can detect that it's explicit cross join because here inspected object is SqlNode, but later after conversion to RelNode this `JoinType.CROSS` will be lost, instead we only have `JoinRelType.INNER` in the node. So to preserve original type additional tricks with query context are necessary. 
What you suggest is actually **Option 2** from my comment in [DRILL-786](https://issues.apache.org/jira/browse/DRILL-786) , but as I remember we agreed to go with **Option 3** .  So what are the reasons to move back and use Option 2  ?